llvm/lib/Target/ARM/ARMAsmPrinter.cpp

```diff
@@ -629,6 +629,15 @@ static bool checkDenormalAttributeConsistency(const Module &M,
   });
 }
 
+// Returns true if all functions have different denormal modes.
+static bool checkDenormalAttributeInconsistency(const Module &M) {
+  if (M.functions().empty())
+    return false;
+  DenormalMode Value = M.functions().begin()->getDenormalModeRaw();
+  return any_of(
```

Comment:
Should only do the any_of from ++begin to avoid re-checking the first function (can also fuse that with the empty check)